tornettools/generate_tgen.py

```diff
@@ -328,12 +371,24 @@ def __get_perf_clients(args, n_users):
     perf_clients = []
 
     n_perf = round(n_users * args.torperf_scale)
-    for i in range(n_perf):
+
+    # split the perf clients between onion-service clients and non-onion-service (public) clients
+    n_hs_perf = round(n_perf * args.onion_service_traffic_scale)
+    n_public_perf = n_perf - n_hs_perf
+
+    for (i, is_hs_client) in ((x, x >= n_public_perf) for x in range(n_perf)):
         chosen_country_code = choice(ONIONPERF_COUNTRY_CODES)
         client = {
             'name': 'perfclient{}'.format(i+1),
             'country_code': chosen_country_code,
+            'is_hs_client': is_hs_client,
         }
+
+        if is_hs_client:
+            client['name'] = 'perfclient{}onionservice'.format(i+1)
+        else:
+            client['name'] = 'perfclient{}'.format(i+1)
```

Comment:
Same as above, do we want `perfclient{}public`? I'm not sure if that will mess up the parsing and plotting scripts though...

Reply:
Renamed the `perfclient{}` hosts to `perfclient{}public`. I made a small change to the tgentools match expression `perfclient[0-9]+(public|onionservice)?\.tgen\.[0-9]+.stdout` and it seems like it works.